[IMP] barcode: New images, style, rewrite for 18.0 (#13020)
odoo/documentation: content/applications/inventory_and_mrp/barcode/setup/software.rst

learn about optional conventions around barcodes and default values in Odoo

To change the barcode nomenclature, go to :menuselection:`Inventory app --> Configuration -->
Settings` and scroll down to the :guilabel:`Barcode` section to select the nomenclature under
:guilabel:`Barcode Scanner`.

.. image:: software/barcodes-setup-change-nomenclature.png
   :alt: Changing barcode nomenclature in Inventory app settings.

.. _barcode/setup/barcodelookup:

Barcode lookup
--------------

Odoo can automatically add product information using the :guilabel:`Stock Barcode Database` setting
for any :abbr:`UPC (Universal Product Code)`, :abbr:`EAN (European Article Number)`, or :abbr:`ISBN
(International Standard Book Number)` barcode.

To enable automatic barcode look up, go to :menuselection:`Inventory app --> Configuration -->
Settings` and scroll down to the :guilabel:`Barcode` section to tick the box for :guilabel:`Stock
Barcode Database`.

.. note::
   Databases hosted on **Odoo.sh** or **on-premise**, require :ref:`configuring an API key
   <barcodelookup/configuration>`.

.. _inventory/barcode/set-barcodes:

> Set Product Barcodes

Author: I think you're right that it's going to read as off in our docs, though, regardless, so lowercasing and declaring that it's about the abstract concept of setting barcodes for products

====================

Barcodes can be assigned to existing products from the Product Barcodes view or from any product
configuration form in the **Inventory**, **Manufacturing** or **Purchase** apps. The barcode field
can be populated either by typing or using scanner input.

When adding a new product, the :ref:`barcode lookup feature <barcode/setup/barcodelookup>` can be
used to automatically find information about a product based on its barcode, and new products can be
added to the database directly from the **Barcode** app by scanning a barcode.

From a product form
-------------------

Barcodes can be added to both existing products and new products as they are created through the
products configuration form. To access a product's form, go to :menuselection:`Inventory app -->
Products --> Products` and select the product to add a barcode to.

In the :guilabel:`General Information` tab, click the :guilabel:`Barcode` field to either type in
the barcode or use a scanner to input the barcode value.

.. image:: software/barcode-add-to-product-form.png

>    :alt: The 'Barcode' field on a product form with an active cursor.

Author: Leaving to find out

.. note::
   If using :doc:`product variants <../../../sales/sales/products_prices/products/variants>`,
   configure barcodes on individual variants and not the product template to allow scanning to
   retrieve the variants.

From Inventory settings
-----------------------

To access the Product Barcodes view, go to :menuselection:`Inventory app --> Configuration -->
Settings --> Configure Product Barcodes`. From this list view, click the :guilabel:`Barcodes` column
for any product to enter its barcode. Barcode scanners will populate this field when the product is
scanned.

.. image:: software/product-barcodes-configure.png
   :alt: Selecting the Barcode field in the Product Configuration list.

> .. tip::

Author: Unfortunately it doesn't update once you add a barcode, but it does help at first! I was fully prepared for that to get cut for being too essentialsy, but I had also already taken the screenshot, so I wrote it up. It is better to have written and ~~lost~~cut than to never have written at all.🥀

format, create a new product definition in the Odoo database with the available information, and add
that product to the receipt to track the quantity on hand in inventory.

.. image:: software/barcode-scan-for-new-product.png
   :alt: New product confirmation dialog.

.. note::
   Even if a transfer order with a product created through **Barcode** is cancelled, the product
   remains available in the products view of **Inventory** unless it is deleted.

Print Location Barcodes
=======================

Barcodes can be assigned to locations to track of where products are stored and manage transfers,
and are automatically available if the :doc:`Storage Locations
<../../inventory/warehouses_storage/inventory_management/use_locations>` feature is enabled.

To print bar codes for locations, go to :menuselection:`Inventory app --> Configuration -->
Settings`, scroll down to the :guilabel:`Warehouse` section and click :guilabel:`Locations`. Tick
the boxes for locations and the :guilabel:`Print` button will appear, downloading a PDF with
barcodes for all selected locations.

.. image:: software/print-storage-location-barcodes.png

>    :alt: Multiple storage locations selected with 'Print' button at the top of the view.

Reviewer: Wanted to shout out that this is a very user-centric and user-emphatic addition to the doc :blush:

used to automatically find information about a product based on its barcode, and new products can be
added to the database directly from the **Barcode** app by scanning a barcode.

From a product form
-------------------

Barcodes can be added to both existing products and new products as they are created through the
products configuration form. To access a product's form, go to :menuselection:`Inventory app -->
Products --> Products` and select the product to add a barcode to.

In the :guilabel:`General Information` tab, click the :guilabel:`Barcode` field to either type in
the barcode or use a scanner to input the barcode value.

.. image:: software/barcode-add-to-product-form.png
   :alt: The 'Barcode' field on a product form with an active cursor.

.. note::
   If using :doc:`product variants <../../../sales/sales/products_prices/products/variants>`,
   configure barcodes on individual variants and not the product template to allow scanning to
   retrieve the variants.

From Inventory settings
-----------------------

To access the *Product Barcodes* configuration page, go to :menuselection:`Inventory app -->

> Configuration --> Settings`. In the Barcode section, under the Barcode Scanner feature, click

Reviewer: ```suggestion
Configuration --> Settings`. In the :guilabel:`Barcode` section, under the :guilabel:`Barcode Scanner` feature, click
```

:icon:`fa-arrow-right` :guilabel:`Configure Product Barcodes`. From this list view, click the
:guilabel:`Barcodes` column for any product to enter its barcode. Barcode scanners will populate
this field when the product is scanned.

.. image:: software/product-barcodes-configure.png
   :alt: Selecting the Barcode field in the Product Configuration form.

.. tip::
   To filter for the products that do not have barcodes yet, click the :icon:`fa-sort-desc`
   :guilabel:`(Toggle Search Panel)` icon to add a Custom Filter where the :guilabel:`Barcode`
   property is :guilabel:`is not set`.

   .. image:: software/barcode-filter-for-no-barcode.png
      :alt: 'Add Custom Filter' pop-up with 'Barcode is not set' configured.

.. _barcode/setup/location:

> From Barcode app

Reviewer 2: Consider *leading* with this instruction and gracefully falling back to the alternative options after, by order of manual entry. It gives readers the option to skip redundant methods after leading with the most relevant one.

----------------

New products definitions can be added to the database product lists as well as having their on-hand
inventory tracked from the **Barcode** app, if the :ref:`Barcode Lookup feature
<barcode/setup/barcodelookup>` is enabled. To create a new receipt for scanning new products, choose
one of two methods:

#. From the **Barcode** front page, tap :guilabel:`Operations`, tap :guilabel:`Receipts`, and then
   tap the :guilabel:`New` button.

> #. Scan a printed :guilabel:`Receipts (WHIN)` inventory command barcode.

Reviewer 2: Think we're missing a '/' here in the parenthesis.
The language involving this step isn't clear to me; is it saying to simply scan the product's barcode label? If so, consider simplifying the instruction.
```suggestion
#. Scan a printed :guilabel:`Receipts (WH/IN)` inventory command barcode.
```

Author: I drafted a [doc](https://github.com/odoo/documentation/blob/18.0-barcode-operation-barcodes-stul/content/applications/inventory_and_mrp/barcode/setup/operation_types.rst) about what I'm talking about since we don't have anything yet. It needs to be `WHIN` but after today I have finally figured out that we're calling that an "operation" and "command" is for the more granular pick/pack/scrap type tasks.

Scanner` feature, click :icon:`fa-arrow-right` :guilabel:`Configure Product Barcodes`. From this
list view, click the :guilabel:`Barcodes` column for any product to enter its barcode. Barcode
scanners will populate this field when the product is scanned.

.. image:: software/product-barcodes-configure.png
   :alt: Selecting the Barcode field in the Product Configuration form.

.. tip::
   To filter for the products that do not have barcodes yet, click the :icon:`fa-sort-desc`

>    :guilabel:`(Toggle Search Panel)` icon to add a Custom Filter where the :guilabel:`Barcode`
>    property is :guilabel:`is not set`.

Reviewer 3: Need to address the capitalized text here, can either remove the capitalization or explicitly state the UI option with a guilabel (but this reads a little funny)
```suggestion
   :guilabel:`(Toggle Search Panel)` icon to add a custom filter where the :guilabel:`Barcode`
   property is :guilabel:`is not set`.
```
OR
```suggestion
   :guilabel:`(Toggle Search Panel)` icon to :guilabel:`Add Custom Filter` where the
   :guilabel:`Barcode` property is :guilabel:`is not set`.
```

> exist, with a button option to :guilabel:`Create New Product`. Pressing this button with search the

Reviewer 2: typo?
```suggestion
exist, with a button option to :guilabel:`Create New Product`. Pressing this button will search the
```

   :alt: New product confirmation dialog.

.. note::
   Even if a transfer order with a product created through **Barcode** is cancelled, the product
   remains available in the products list in the **Inventory** app unless it is deleted.

> Print Location Barcodes

Reviewer: ```suggestion
Print location barcodes
```
To match the product one above (:

> Print location barcodes

Reviewer 2: This is a good start. 
Printing, in my mind, would come *after* *Location* configuration for products based on their individual form values, product categories, lots, etc. That information would be good to include here, using the barcode scanner to help, i.e., receive new products not previously configured in Odoo --> point the scanner at the Lot of products --> batch scan --> programs their location destination in Odoo as well as their inventory counts (something like this).
